samples/javascript_typescript/60.multilingual-luis-bot/translator.ts

```diff
@@ -0,0 +1,345 @@
+            let texts = options.texts;
+            let orgTexts = [];
+            let translationRequests = [];
+            let results = [];
+            texts.forEach((text, index, array) => {
+                let normalizedText = text.trim();
+                    orgTexts.push(normalizedText);
+                    texts[index] = normalizedText;
+                    let translationRequest = new TranslationRequest();
+                    translationRequest.Text = normalizedText;
+                    translationRequests.push(translationRequest);
+            });
+            let uri: any = `https://api.cognitive.microsofttranslator.com/translate?api-version=3.0&includeAlignment=true&includeSentenceLength=true&from=${from}&to=${to}`;
+			const response = await request({
+				url: uri,
+				method: 'POST',
+				headers: {
+                    'Ocp-Apim-Subscription-Key': this.apiKey,
+					'Content-Type': 'application/json',
+				},
+				body: JSON.stringify(translationRequests),
+            });
+            let translations = JSON.parse(response);
+            translations.forEach((traslationModel, index, array) => {
+                                let translation = traslationModel.translations[0].text;
+                                if(traslationModel.translations[0].alignment){
+                                    let alignment = traslationModel.translations[0].alignment.proj;
+                                    translation = this.postProcessor.fixTranslation(orgTexts[index], alignment, translation, from);
+                                }
+                                let result: TranslationResult = { translatedText: translation };
+                                 results.push(result);
+                            });
+			return results;
+		} catch (e) {
+			switch (e.statusCode) {
+				case 401:
+					throw new Error('Cognitive Authentication Failed');
+				default:
+					throw new Error('Internal Error');
+			}
+		}
+    };
+}
+
+
+/**
+ * @private
+ */
+class TranslatorPostProcessor {
+    noTranslatePatterns: { [languageId : string] : string[]};
+    wordDictionary: { [id: string]: string };
+
+    constructor(noTranslatePatterns?: { [languageId : string] : string[]}, wordDictionary?: { [id: string]: string }) {
+        this.noTranslatePatterns = {};
+        this.wordDictionary = wordDictionary;
+        
+        if (wordDictionary) {
+            Object.keys(this.wordDictionary).forEach(word => {
+                if (word != word.toLowerCase()) {
+                    Object.defineProperty(this.wordDictionary, word.toLowerCase(),
+                        Object.getOwnPropertyDescriptor(this.wordDictionary, word));
+                    delete this.wordDictionary[word];
+                }
+            });
+            
+        }
+
+        if(noTranslatePatterns) {
+            for(var key in noTranslatePatterns) {
+                noTranslatePatterns[key].forEach((pattern, index, array) => {
```

Comment:
```suggestion
                noTranslatePatterns[key].forEach((pattern, index) => {
```
Remove unused variable